.github/workflows/test_conda_store_server_unit.yaml

```diff
@@ -0,0 +1,79 @@
+name: "Test conda-store-server (unit)"
+
+env:
+  DEFAULT_PYTHON_VERSION: "3.10"
+  FORCE_COLOR: "1" # Make tools pretty.
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+
+# ensuring only one instance is running at a given time
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test-conda-store-server:
+    name: "unit-test conda-store-server"
+    strategy:
+      matrix:
+        os: ["ubuntu", "macos", "windows"]
+        include:
+          - os: ubuntu
+            environment-file: conda-store-server/environment-dev.yaml
+          - os: macos
+            environment-file: conda-store-server/environment-macos-dev.yaml
+          - os: windows
+            environment-file: conda-store-server/environment-windows-dev.yaml
+    runs-on: ${{ matrix.os }}-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+        working-directory: conda-store-server
+    steps:
+      - name: "Checkout Repository 🛎"
+        uses: actions/checkout@v4
+
+      - name: "Set up env ${{ matrix.os }} 🐍"
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+            environment-file: ${{ matrix.environment-file }}
+            miniforge-version: latest
+
+      # This fixes a "DLL not found" issue importing ctypes from the hatch env
+      - name: "Reinstall Python 3.10 on Windows runner"
+        uses: nick-fields/retry@v2.8.3
+        with:
+          timeout_minutes: 9999
```

Comment:
I know this isn't your change, but does it make sense to put something reasonable here?

Reply:
I remember when this was added and had concerns as well. But I think it's fine. The goal of this timeout is to make sure it never hits before `max_attempts: 6` below executes, so it's less flaky this way. It's been working fine, so I see no reason to change this.